test/demo-extension/contentscript/api.ts

```diff
@@ -0,0 +1,10 @@
+import { getContentScriptMethod } from "../../../index";
+
+export const getPageTitle = getContentScriptMethod("getPageTitle");
+export const setPageTitle = getContentScriptMethod("setPageTitle");
+export const closeSelf = getContentScriptMethod("closeSelf");
+export const sumIfMeta = getContentScriptMethod("sumIfMeta");
+export const contentScriptOnly = getContentScriptMethod("contentScriptOnly");
+export const throws = getContentScriptMethod("throws");
+export const notRegistered = getContentScriptMethod("notRegistered");
+export const getSelf = getContentScriptMethod("getSelf");
```

Comment:
As mentioned in a previous PR, some errors are just undetectable, however we slice it. For example here I'm registering a `getSelf` method for the content script, even if there's already another (identical) function registered in the background.

No API/type will guarantee that there's no `getSelf` method registered in multiple contexts or that they have the same implementation, **but**:

- the current setup _does_ ensure that `getSelf` has at least the same signature, so it will not let you register 2 homonymous methods with differing signatures (which partially avoids the problem)
- even if there are 2 homonymous methods, it's all good as long as they're registered in different contexts, because the message will always travel to a specific context
- Since I'm using globals (and not instances, in any case), `registerMethods` will not let you register the same name twice, so it's impossible to register 2 methods in the same context with the same name.